src/parser/Parser.ts

```diff
@@ -2418,17 +2466,70 @@ export class Parser {
         }
     }
 
+    private getBscTypeFromAssignment(assignment: AssignmentStatement, functionExpression: FunctionExpression): BscType {
+        try {
+            //function
+            if (isFunctionExpression(assignment.value)) {
+                let functionType = new FunctionType(assignment.value.returnType);
+                functionType.isSub = assignment.value.functionType.text === 'sub';
+                if (functionType.isSub) {
+                    functionType.returnType = new VoidType();
+                }
+
+                functionType.setName(assignment.name.text);
+                for (let param of assignment.value.parameters) {
+                    let isRequired = !param.defaultValue;
+                    //TODO compute optional parameters
+                    functionType.addParameter(param.name.text, param.type, isRequired);
+                }
+                return functionType;
+
+                //literal
+            } else if (isLiteralExpression(assignment.value)) {
+                return assignment.value.type;
+
+                //function call
+            } else if (isCallExpression(assignment.value)) {
+                let calleeName = ((assignment.value.callee as any).name.text as string).toLowerCase();
+                if (calleeName) {
+                    //make a lazy type which will not compute its type until the file is done parsing
+                    return new LazyType(() => {
+                        //TODO this should really be scope-aware...
+                        const func = this.references.functionStatements.find(x => x.name.text.toLowerCase() === calleeName);
+                        return func?.func.returnType ?? new DynamicType();
+                    });
+                }
+            } else if (isVariableExpression(assignment.value)) {
+                //defer the type until first read, which will allow us to derive types from variables defined after this one (like from a loop perhaps)
+                return new LazyType(() => {
```

Comment:
Same as before. I think `LazyTypes` should be more of a "scope-aware" callable resolver, or something we keep track of in references and resolve as part of scope validation (that's what I do in bslint variable tracking logic).